libs-scala/db-utils/src/main/scala/com/digitalasset/http/dbbackend/DBConfig.scala

```diff
@@ -37,31 +35,41 @@ abstract class ConfigCompanion[A, ReadCtx](name: String) {
 
   protected val indent: String = List.fill(8)(" ").mkString
 
-  protected[this] def create(x: Map[String, String], defaultDriver: Option[String])(implicit
+  // If we don't DRY our keys, we will definitely forget to remove one.  We're
+  // less likely to make a mistake in backend-specific conf if redundant data
+  // isn't there. -SC
+  type Fields[Z] = StateT[({ type l[a] = Either[String, a] })#l, Map[String, String], Z]
```

Comment:
hey @S11001001 I am not overtly familiar with the use of `scalaz.StateT` in general so I am kinda interested in understanding the reasoning for the use of `StateT` here and why maintaining state (i.e removing keys from the map) makes it more robust or intuitive ( coming from a vanilla scala side I found the previous code more intuitive)

Could you perhaps please explain with a couple of lines or point me to any docs/blog for this type of usage ?

Reply:
I want to remove keys from the custom "backend-specific" config that are used by the agnostic config. Fewer possible inputs, fewer situations to think about.

The alternative here is to have a separate list of the keys we use, and remove them all at once when filling in the `backendSpecificConf`. I simply don't believe that we could possibly successfully maintain that list of keys accurately. So I made it so that we had to get it right, constructively, instead.